### ISSUE TITLE
v4.0.0-beta.10 - getAccessToken in api handler raises Invalid URL

### ISSUE THREAD

**Issue by chris-erickson** [auth0/nextjs-auth0#1843]
### Checklist

- [X] The issue can be reproduced in the [nextjs-auth0 sample app](https://github.com/auth0-samples/auth0-nextjs-samples/tree/main/Sample-01) (or N/A).
- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

Updating our app to v4, mostly going well. We have a "/user/me" route that returns user data from our internal database, using the access token. It's currently raising an error: 
```
TypeError: Failed to parse URL from /auth/access-token
[cause]: TypeError: Invalid URL
    code: 'ERR_INVALID_URL',
    input: '/auth/access-token'
```

### Reproduction

Some possibly relevant env settings:
```
APP_BASE_URL="http://localhost:3000"
AUTH0_DOMAIN="dev-our-id-here.us.auth0.com"
```

Example of failing code:
```typescript
// app/api/user/me/route.ts
import { getAccessToken } from "@auth0/nextjs-auth0";

export const GET = async () => {
  const token = await getAccessToken(); // <-- Raises error
```

### nextjs-auth0 version

v4.0.0-beta.10

### Next.js version

15.1.0

### Node.js version

v20.17

**Comment by chris-erickson** [comment]
Interestingly [this solution appears to solve it](https://community.auth0.com/t/await-getaccesstoken-fails-with-typeerror-failed-to-parse-url-from-auth-access-token/152109/2), but is at odds with the v4 Migration Guide, which plainly says getAccessToken works in RSC components now but doesn't suggest any other migrations.

```typescript
const session = await auth0.getSession()
const accessToken = session?.tokenSet.accessToken
```

**Comment by jdwitten** [comment]
@chris-erickson Try using the `auth0` client the migration guide suggests to initialize in `/src/lib/auth0`. I believe I ran into a similar issue but once I did this, things appeared to start working:
```
import { auth0 } from './src/lib/auth0'

...
const { token } = await auth0.getAccessToken()
```

Would be good to have this documented if this was intended to be a breaking change

**Comment by chris-erickson** [comment]
I wonder which is best. I need to check the session anyways so is my option, while kind of ugly.. simpler? Good to know I could probably just update the call to `getAccessToken` as you suggest though.

**Comment by ajwootto** [comment]
I think the `getAccessToken` method that is directly exported from the SDK is only meant to be used clientside. It should probably have a proper error if executed on the server.

It also seems like a big footgun to me to have a directly exported method `getAccessToken` that is named the same as the method on the exported auth0 instance, but only works clientside vs serverside. 

Seems like the client should only provide a hook like `useAccessToken` to make it clear that it only works in clients (and then use SWR under the hood to store the fetched token), or perhaps name the method `getAccessTokenClientside` or something.

**Comment by guabu** [comment]
Hey @chris-erickson 👋 The solutions suggested by @jdwitten and @ajwootto are right — since you're calling the `getAccessToken()` method on the server, you will need to use the server instance of the Auth0Client initialized in `./lib/auth0`.

Thanks for the feedback here @ajwootto. I agree, we can definitely provide clearer error messages in this case — we'll get that updated in the next release. We've gone back and forth on the naming quite a bit, perhaps it would make sense to export it under `/client` the same way we do for the `/server` exports to be more explicit.

**Comment by chris-erickson** [comment]
Just seeing in the v4 readme [the import statement is confusing](https://github.com/auth0/nextjs-auth0/tree/v4?tab=readme-ov-file#in-the-browser-1) (and subsequent usage example):

```javascript
import { getAccessToken } from "@auth0/nextjs-auth0"

export default function Component() {
  async function fetchData() {
    try {
      const token = await auth0.getAccessToken()
```

**Comment by nandan-bhat** [comment]
Closing this issue as it has been resolved in `4.0.0-beta.11`.
